firmware/library/L3_HAL/sddriver.hpp

```diff
@@ -0,0 +1,239 @@
+#pragma once
+
+#include <cstdint>
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/ssp.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+// Relevant Datasheet: https://www.sdcard.org/downloads/pls/pdf/index.php?p=Part1_Physical_Layer_Simplified_Specification_Ver6.00.jpg&f=Part1_Physical_Layer_Simplified_Specification_Ver6.00.pdf&e=EN_SS1
+class SdInterface
+{
+ public:
+    // // @constructor     SdInterface()
+    // constexpr SdInterface() = 0;
+
+    // @destructor      ~SdInterface
+    virtual ~SdInterface() {};
+
+    // @struct          SdResponse_t
+    // @description     (see below)
+    // TODO(#151): Define a response frame struct since there are various
+    // responses that will be sent, depending on the command sent previously
+    struct SdResponse_t
+    {
+        union {
+            uint64_t qWord;
+            struct {
+                uint32_t hi;
+                uint32_t lo;
+            } dWord;
+            uint8_t byte[8];
+        } data;
+        uint8_t length;
+    };
+
+    // @enum class      SdResponseType
+    // @description     Enumerator for response type codes
+    enum class SdResponseType
+    {
+        kR1,
+        kR1b,
+        kR2,
+        kR3,
+        kR4,
+        kR5,
+        kR6,
+        kR7
+    };
+
+    // @enum class      SdType
+    // @description     Enumerator for SD Card Capacity type codes
+    enum class SdType
```

Comment:
Rather than adding a comment to explain what this does, you can change the name of the enumeration. For example, you could name it `Capacity`. Then it is obvious by the name, that its an enumeration for the SD card's capacity.